game/layout/layoutloader.py

```diff
@@ -0,0 +1,203 @@
+        mappings: dict[str, list[LayoutMapping]] = {}
+        with logged_duration("Parsing mapping yamls"):
+            for file in Path(TEMPLATE_DIR).rglob("*.yaml"):
+                if not file.is_file():
+                    continue
+                with file.open("r", encoding="utf-8") as f:
+                    mapping_dict = yaml.safe_load(f)
+
+                template_map = LayoutMapping.from_dict(mapping_dict, f.name)
+
+                if template_map.layout_file in mappings:
+                    mappings[template_map.layout_file].append(template_map)
+                else:
+                    mappings[template_map.layout_file] = [template_map]
+
+        with logged_duration(f"Parsing all layout miz multithreaded"):
+            with ThreadPoolExecutor() as exe:
+                for miz, maps in mappings.items():
+                    exe.submit(self._load_from_miz, miz, maps)
+
+        logging.info(f"Imported {len(self._templates)} layouts")
+        self._dump_templates()
+
+    def _dump_templates(self) -> None:
+        file = Path(persistency.base_path()) / TEMPLATE_DUMP
+        dump = (VERSION, self._templates)
+        with file.open("wb") as fdata:
+            pickle.dump(dump, fdata)
+
+    @staticmethod
+    def mapping_for_group(
+        mappings: list[LayoutMapping], group_name: str
+    ) -> tuple[LayoutMapping, int, GroupLayoutMapping]:
+        for mapping in mappings:
+            for g_id, group_mapping in enumerate(mapping.groups):
+                if (
+                    group_mapping.name == group_name
+                    or group_name in group_mapping.statics
+                ):
+                    return mapping, g_id, group_mapping
+        raise KeyError
+
+    def _load_from_miz(self, miz: str, mappings: list[LayoutMapping]) -> None:
+        template_position: dict[str, Point] = {}
+        temp_mis = dcs.Mission()
+        with logged_duration(f"Parsing {miz}"):
+            # The load_file takes a lot of time to compute. That's why the layouts
+            # are written to a pickle and can be reloaded from the ui
+            # Example the whole routine: 0:00:00.934417,
+            # the .load_file() method: 0:00:00.920409
+            temp_mis.load_file(miz)
+
+        for country in itertools.chain(
+            temp_mis.coalition["red"].countries.values(),
+            temp_mis.coalition["blue"].countries.values(),
+        ):
+            for dcs_group in itertools.chain(
+                temp_mis.country(country.name).vehicle_group,
+                temp_mis.country(country.name).ship_group,
+                temp_mis.country(country.name).static_group,
+            ):
+                try:
+                    mapping, group_id, group_mapping = self.mapping_for_group(
+                        mappings, dcs_group.name
+                    )
+                except KeyError:
+                    logging.warning(f"No mapping for dcs group {dcs_group.name}")
+                    continue
```

Comment:
true that. it was the other way around to prevent performance issues as in the previous version this was only one function iterating over all miz files. There it would open all miz files n times where n is the number of available mappings. As this function was cut down this makes more sense to loop through all mappings and look for them as the perfomance intensive zip opening already happened